### PR TITLE
Separate advanced cost model from the previous basic cost model

### PR DESCRIPTION
**Pull request by jurplel**
Tested only briefly, quick and dirty

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,6 @@ members = [
     "optd-sqlplannertest",
     "optd-adaptive-demo",
     "optd-gungnir",
-    "optd-perfbench",
+    "optd-perfbench", "optd-datafusion-repr-adv-cost",
```
Comment (skyzh) [inline Cargo.toml]: fmt?

File: optd-datafusion-repr-adv-cost/src/bin/test_optimize_adv.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+use std::{collections::HashMap, sync::Arc};
```
Comment (skyzh) [inline optd-datafusion-repr-adv-cost/src/bin/test_optimize_adv.rs:1]: make it a unit test? (can be done later)

**Review by skyzh — APPROVED** [review]
need to run regression tests, likely a lot of plans have changed

**Comment by jurplel** [comment]
An argument could be made I think to separate the ColumnRef property too, but i'll leave it for now

**Pull request merged**